src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,834 @@
+}
+
+let gExtension;
+
+function sha1(str) {
+  console.log(`string at start : ${str}`);
+  let converter =
+    Components.classes["@mozilla.org/intl/scriptableunicodeconverter"].
+      createInstance(Components.interfaces.nsIScriptableUnicodeConverter);
+  converter.charset = "UTF-8";
+  let result = {};
+  let data = converter.convertToByteArray(str, result);
+  console.log(`data result = ${str}`);
+  let ch = Components.classes["@mozilla.org/security/hash;1"]
+                     .createInstance(Components.interfaces.nsICryptoHash);
+  ch.init(ch.SHA1);
+  ch.update(data, data.length);
+  console.log(`ch = ${ch}`);
+  let hash = ch.finish(false);
+  console.log(`hash = ${hash}`);
+  function toHexString(charCode) {
+    return ("0" + charCode.toString(16)).slice(-2);
+  }
+  console.log(`returned value = ${Array.from(hash, (c, i) => toHexString(hash.charCodeAt(i))).join("")}`)
+  return Array.from(hash, (c, i) => toHexString(hash.charCodeAt(i))).join("");
+}
+
+function isEmailValid(val) {
+  // https://stackoverflow.com/a/46181
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(val).toLowerCase());
+};
+
+function showInvalidMessage(textbx) {
+  textbx.style.borderStyle = "solid";
+  textbx.style.borderColor = "#d70022cc";
+  textbx.style.borderWidth = "1px";
+  textbx.value = "";
+  textbx.placeholder = "Please enter a valid email.";
+  textbx.boxShadow = "1px 0px 4px #d700224d";
+  textbx.style.transition = "all 0.2s ease";
+}
+
+function clearInvalidMessage(textbx) {
+  textbx.style.borderStyle = "solid";
+  textbx.style.borderColor = "rgba(12, 12, 13, 0.30)";
+  textbx.style.borderWidth = "1px";
+  textbx.placeholder = "Please enter a valid email.";
+  textbx.boxShadow = "1px 0px 4px rgba(12, 12, 13, 0.09)";
+  textbx.transition = "all 0.2s ease";
```

Comment:
Should this be `textbx.style.transition` to match L49 above?

Reply:
yes, and same thing for boxShadow!